### PR TITLE
[API Compatibility No.155] Add kwargs alias for binary_cross_entropy_with_logits-part

### PR DESCRIPTION
**Pull request by Aidenwu0209**
### PR Category
User Experience

### PR Types
New features

### Description
- For `paddle.nn.functional.binary_cross_entropy_with_logits`, add kwargs alias compatibility with PyTorch:
  - `input` as alias of `logit`
  - `target` as alias of `label`
- Implemented with `@param_two_alias([logit, input], [label, target])`.
- Updated docstring alias notes for `logit` and `label`.
- Added legacy test `test_binary_cross_entropy_with_logits_alias` to cover:
  - equivalence of (`logit`, `label`) vs (`input`, `target`) for `reduction=none` and `mean`
  - `ValueError` when both canonical and alias kwargs are passed together.
- Related issue: #76301 (No.155)

### 是否引起精度变化
否

### CODE REVIEW

**Review by zhwesky2010 — DISMISSED** [review]
LGTM

**Comment by zhwesky2010** [comment]
loss这一系列API后面统一提交一个paconvert测试PR吧，需要paconvert全部通过，同时需要增加size_average/reduce的测试。

File: test/legacy_test/test_bce_with_logits_loss.py [hunk 1]
```diff
@@ -303,6 +303,46 @@ def test_static_or_pir_mode():
 
         test_static_or_pir_mode()
 
+    def test_binary_cross_entropy_with_logits_alias(self):
+        paddle.disable_static()
+        try:
```
Comment (zhwesky2010) [inline test/legacy_test/test_bce_with_logits_loss.py]: 测试代码不要使用try（除非是专门测异常抛出的），需要保证测试case正确可运行

Comment (Aidenwu0209) [reply test/legacy_test/test_bce_with_logits_loss.py]: 已修改，去掉 try/finally，改为 `self.addCleanup(paddle.enable_static)` 保证测试环境恢复。修复提交：1521b15，请帮忙再看下。

**Comment by Aidenwu0209** [comment]
已按 review 修改：去掉测试中的 try/finally，改为 self.addCleanup(paddle.enable_static)（commit: 1521b15）。\n另外，您提到的 loss 系列 PaConvert 测试已在 PR https://github.com/PaddlePaddle/PaConvert/pull/846 补齐并包含 size_average/reduce 相关用例。\n@zhwesky2010 CI 跑完后麻烦再帮忙看下。

**Review by zhwesky2010 — APPROVED** [review]
LGTM

**Review by YqGe585 — APPROVED** [review]
LGTM for IXUCA
